modules/microshift-proc-configuring-generic-device-plugin.adoc

```diff
@@ -0,0 +1,131 @@
+// Module included in the following assemblies:
+//
+// microshift_configuring/microshift-gdp.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-configuring-generic-device-plugin_{context}"]
+= Configuring the Generic Device Plugin
+
+:FeatureName: The Generic Device Plugin for {microshift-short}
+
+include::snippets/technology-preview.adoc[]
+
+The Generic Device Plugin (GDP) is disabled by default in {microshift-short}. To use the GDP, you must enable it and specify which host devices your Kubernetes applications can access. To enable the GDP, you must modify the {microshift-short} `config.yaml` file or create a configuration drop-in file (for example, `/etc/microshift/config.d/10-gdp.yaml`).
+
+.Prerequisites
+
+* You installed {microshift-short}.
+* You created a custom `config.yaml` file in the `/etc/microshift` directory.
+* You installed the {oc-first}.
+* You have `sudo` privileges on the {microshift-short} host.
+* You have identified the specific host devices that you want to expose to your {microshift-short} cluster (for example, `/dev/video0`, `/dev/ttyUSB*`, or USB Vendor/Product IDs).
+
+.Procedure
+
+. Open `/etc/microshift/config.yaml` in a text editor with `sudo` privileges.
++
+. Locate the `genericDevicePlugin` section. If it is not present, add it.
++
+. Set the `status` parameter to `Enabled` and define the `devices` that should be exposed. Each device definition needs a `name` and one or more `groups`. Each group can specify devices using `paths`, for file-based devices, including glob patterns, or `usbs`, for USB devices using Vendor/Product IDs. You cannot mix `paths` and `usbs` within the same device group.
++
+.GDP fields with default values
+[source,yaml]
+----
+apiServer:
+# ...
+genericDevicePlugin:
+  status: Enabled
+  domain: device.microshift.io <1>
+  devices:
+    - name: serial
+      groups:
+        - paths:
+            - path: /dev/ttyUSB* <2>
+            - path: /dev/ttyACM* <3>
+    - name: fuse
+      groups:
+        - paths:
+            - path: /dev/fuse
+    - name: converter
+      groups:
+        - usbs: <4>
+            - vendor: "0x1a86" <5>
+              product: "0x7523" <6>
+----
+<1> Default domain for the GDP.
+<2> Exposes all the USB serial devices that are matched by this glob.
+<3> Exposes all the ACM serial devices that are matched by this glob.
+<4> Exposes a specific USB device by Vendor ID and Product ID.
+<5> For example, the Vendor ID for a CH340 serial converter.
+<6> For example, the Product ID for a CH340 serial converter
++
+[IMPORTANT]
+====
+* The output of the `microshift show-config` parameter might include pre-configured default paths for serial, video, and audio devices even if you have not explicitly configured them in `config.yaml`. These paths represent the default discovery settings if the Generic Device Plugin is enabled without specific user configuration.
```

Comment:
We do not see serial, video and audio devices, i only see for serial devices, see below

```
genericDevicePlugin:
  devices:
  - groups:
    - count: 1
      paths:
      - limit: 1
        mountPath: /dev/ttyACM0
        path: /dev/ttyACM0
        permissions: mrw
        type: Device
    name: serial
  domain: device.microshift.io
  status: Enabled

```